internal/core/data/event.go

```diff
@@ -192,37 +197,49 @@ func getEventById(id string) (contract.Event, error) {
 	return e, nil
 }
 
+// updateEventPushDateByChecksum updates the pushed dated for all events with a matching checksum which have not already been marked pushed
+func updateEventPushDateByChecksum(checksum string, ctx context.Context) error {
+	evts, err := dbClient.EventsByChecksum(checksum)
+	if err != nil {
+		return err
+	}
+
+	for _, e := range evts {
+		e.Pushed = db.MakeTimestamp()
+		// Updating the event has the desired side-effect of removing the checksum.
+		// We only want the checksum for "marked pushed" functionality and once the event
+		// has been marked pushed there is no reason to keep the checksum around.
+		// The expectation is that above query will only return one result, but this is not guaranteed
+		err = updateEvent(models.Event{Event: e}, ctx)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func updateEventPushDate(id string, ctx context.Context) error {
 	e, err := getEventById(id)
 	if err != nil {
 		return err
 	}
 
 	e.Pushed = db.MakeTimestamp()
-	err = updateEvent(e, ctx)
+	err = updateEvent(models.Event{Event: e}, ctx)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
 // Put event on the message queue to be processed by the rules engine
-func putEventOnQueue(e contract.Event, ctx context.Context) {
+func putEventOnQueue(evt models.Event, ctx context.Context) {
 	LoggingClient.Info("Putting event on message queue")
-	//	Have multiple implementations (start with ZeroMQ)
-	evt := models.Event{}
-	evt.Event = e
+	// 	Have multiple implementations (start with ZeroMQ)
```

Comment:
Why is this comment needed. This code is independent of the message bus implementation.

Reply:
Removed